### DISCUSSION TITLE
async IPC command returntype issues

### DISCUSSION THREAD

**Discussion by sasial-dev** [tauri-apps/tauri#5061] [Q&A] [answered]
Hey, I'm getting the following:

> the method `async_kind` exists for reference `&impl Future<Output = Result<String, anyhow::Error>>`, but its trait bounds were not satisfied
> the following trait bounds were not satisfied:
> `impl Future<Output = Result<String, anyhow::Error>>: serde::ser::Serialize`
> which is required by `&impl Future<Output = Result<String, anyhow::Error>>: tauri::command::private::SerializeKind`
> `<&impl Future<Output = Result<String, anyhow::Error>> as Future>::Output = Result<_, _>`
> which is required by `&impl Future<Output = Result<String, anyhow::Error>>: tauri::command::private::ResultFutureKind`
> `&impl Future<Output = Result<String, anyhow::Error>>: Future`
> which is required by `&impl Future<Output = Result<String, anyhow::Error>>: tauri::command::private::ResultFutureKind`

My Code:
```rs
use anyhow::{anyhow, Result};
use rustube::{
    Id,
    VideoFetcher
};

#[tauri::command]
async fn get_video_title(url: String) -> Result<String> {
    let id = Id::from_string(url).map_err(|e| anyhow!("Invalid YouTube Link: {e}"))?;
    Ok(VideoFetcher::from_id(id)?.fetch().await?.video_title().to_owned())
}
```

**Comment by FabianLars** [comment, marked as answer]
For visibility, here are my discord messages: https://discord.com/channels/616186924390023171/1012868459757052048/1013121224450256947
> You can't use anyhow::Result (or its Error) as a return value of tauri commands
> 
> everything you return from commands needs to implement `serde::Serialize`, and anyhow (and most other crates) don't do that by default
>
> Here are our wip docs with a possible solution: <https://jonaskruckenberg.github.io/tauri-docs-wip/development/inter-process-communication.html#error-handling>
